File: backend/app/models/agedatastore.py
from ..db import db
from sqlalchemy import UUID


class Agedatastore(db.Model):
    __tablename__ = 'agedatastores'

    guid = db.Column(UUID, primary_key=True, nullable=False, unique=True)
    name = db.Column(db.String(120), nullable=False)
    alias = db.Column(db.String(120), nullable=False)
    type = db.Column(db.String(120), nullable=False)
    ishosted = db.Column(db.Boolean, nullable=False)
    url = db.Column(db.String(255), nullable=True)
    status = db.Column(db.String(120), nullable=False)
    description = db.Column(db.String(255), nullable=True)
    capacity_gb = db.Column(db.Integer, nullable=False)
    used_gb = db.Column(db.Integer, nullable=False)

    def to_dict(self):
        return {
            "guid": self.guid,
            "name": self.name,
            "alias": self.alias,
            "description": self.description,
            "url": self.url,
            "type": self.type,
            "ishosted": self.ishosted,
            "status": self.status,
            "capacity_gb": self.capacity_gb,
            "used_gb": self.used_gb
        }

    def __repr__(self):
        return f'AGE Data Store {self.name}'

File: frontend/src/models/age-datastore.ts
import {AbstractModel} from "./abstract-models.ts";

export interface AgeDatastoreProps {
    guid: string;
    name: string;
    alias: string;
    type: string;
    ishosted: boolean;
    url: string | null;
    status: string;
    description: string | null;
    capacity_gb: number;
    used_gb: number;
}

export class AgeDataStore extends AbstractModel {
    class: string = 'AgeDataStore';

    guid: string;
    name: string;
    alias: string;
    type: string;
    ishosted: boolean;
    url: string | null;
    status: string;
    description: string | null;
    capacity_gb: number;
    used_gb: number;

    constructor(props: AgeDatastoreProps) {
        super()
        this.guid = props.guid;
        this.name = props.name;
        this.alias = props.alias;
        this.type = props.type;
        this.ishosted = props.ishosted;
        this.url = props.url;
        this.status = props.status;
        this.description = props.description;
        this.capacity_gb = props.capacity_gb;
        this.used_gb = props.used_gb;
    }
}
